### PR TITLE
Expose a Paragraph.getWordBoundary method to dart.

### PR DESCRIPTION
**Pull request by mpcomplete**
Used by text selection to select a word.

### CODE REVIEW

**Comment by Hixie** [comment]
What is a word? Is it a sequence of characters surrounded by line break opportunity transitions? A sequence of characters that we do not consider to be whitespace for the purposes of justification?

**Comment by Hixie** [comment]
LGTM if you add a comprehensive description of what you mean by "word" to the dartdoc.

**Comment by abarth** [comment]
> What is a word?

It's super complicated and the main reason we need to link with the ICU data table.  :)

**Comment by mpcomplete** [comment]
> It's super complicated and the main reason we need to link with the ICU data table. :)

Yeah, I ended up linking to the unicode spec:  http://www.unicode.org/reports/tr29/#Word_Boundaries

File: sky/engine/core/dart/text.dart [hunk 1]
```diff
@@ -674,6 +674,13 @@ abstract class Paragraph extends NativeFieldWrapperClass2 {
     return new TextPosition(offset: encoded[0], affinity: TextAffinity.values[encoded[1]]);
   }
   List<int> _getPositionForOffset(Offset offset) native "Paragraph_getPositionForOffset";
+
+  /// Returns the [start, end] of the word at the given offset. Characters not
+  /// part of a word, such as spaces, symbols, and punctuation, have word breaks
+  /// on both sides. In such cases, this method will return [offset, offset+1].
+  /// Word boundaries are defined more precisely in Unicode Standard Annex #29
+  /// http://www.unicode.org/reports/tr29/#Word_Boundaries
```
Comment (Hixie) [inline sky/engine/core/dart/text.dart]: This is great, thanks!

BTW, I believe this will only linkify if you put the url in angle brackets.

**Pull request merged**